### PR TITLE
Overhaul volto customization training

### PR DESCRIPTION
**Pull request by nileshgulia1**
- Modified Installation docs (Use Cookieplone only)
- Rearranged some chapters and added the latest content.
- Removed Extending teasers per type chapter as its out of scope for this training.

----
📚 Documentation preview 📚: https://plone6--860.org.readthedocs.build/

### CODE REVIEW

**Comment by stevepiercy** [comment]
@nileshgulia1 do you have permission to request a review from other developers?

I have a huge backlog of documentation PRs to review before I can get to this one, but I would like to take a look, if I find time. Thank you!

**Comment by nileshgulia1** [comment]
> @nileshgulia1 do you have permission to request a review from other developers?
> 
> 
> 
> I have a huge backlog of documentation PRs to review before I can get to this one, but I would like to take a look, if I find time. Thank you!

@stevepiercy Although its ready, I would do it during the weekdays. Just don't want to bother you during weekend :). But here you go!

File: docs/volto_customization/installation.md
Comment (sneridagh) [inline docs/volto_customization/installation.md:33]: @nileshgulia1 this is not accurate, 6.0 is "LTS" so the recommended version is Volto 16.0.0 which is also "LTS".

Comment (nileshgulia1) [reply docs/volto_customization/installation.md]: fixed in [d91612d](https://github.com/plone/training/pull/860/commits/d91612d721ed191d3f0935f6a5758754357f06f7)

File: docs/volto_customization/installation.md
Comment (stevepiercy) [inline docs/volto_customization/installation.md:28]: I would suggest that you replace all the above with a link to https://6.docs.plone.org/install/create-project-cookieplone.html#prerequisites-for-installation. This uses Intersphinx and MyST syntax. Then you'll never have to update it again. 😁 

Also it is not good for accessibility to force users to open links in a new browser window, especially for web app developers who know what they are doing while browsing. I personally despise that behavior, and prefer to command-click, right-click and open in new window, or use my mouse's scroll wheel click.

```suggestion
Follow the guide in Plone 6 Documentation, {ref}`create-project-cookieplone-prerequisites-for-installation-label`.
```

File: docs/volto_customization/installation.md
Comment (stevepiercy) [inline docs/volto_customization/installation.md:35]: Plone 5.2 is no longer supported, so we shouldn't encourage its usage in any way.

```suggestion
```

File: docs/volto_customization/installation.md
Comment (stevepiercy) [inline docs/volto_customization/installation.md:43]: Cookieplone is capitalized in narrative text. Please check throughout for proper case.

Also this step is not installation, but generation of a project.

Finally, I think the link at this point is a distraction, as your instructions differ slightly. It would be OK at the end of this section as a `seealso`.

```suggestion
You can use pipx to run Cookieplone to generate a project.
```

File: docs/volto_customization/installation.md
Comment (stevepiercy) [inline docs/volto_customization/installation.md:49]: Hyphenate "add-on" wherever it occurs in narrative text. The Cookieplone prompt does not hyphenate it. I'll fix that in the template.

```suggestion
You will be presented with a series of prompts.
You can also specify the add-ons you want to install along with the project.
When prompted for {guilabel}`Volto Addon Name`, enter `volto-teaser-tutorial`.
```

This was confusing because you say to change two things here, then later to accept the defaults except for one. I edited the inconsistent sentence.

File: docs/volto_customization/installation.md
Comment (stevepiercy) [inline docs/volto_customization/installation.md:62]: This is the default value. Replace all instances throughout the training for consistency.

```suggestion
Change your current working directory to `project-title`.
```

File: docs/volto_customization/installation.md
Comment (stevepiercy) [inline docs/volto_customization/installation.md:65]: Same

File: docs/volto_customization/installation.md
Comment (stevepiercy) [inline docs/volto_customization/installation.md:100]: Line 98 gets converted to a hyperlink automatically. All URLs do with our MyST configuration. I would remove this link.

File: docs/volto_customization/installation.md
Comment (stevepiercy) [inline docs/volto_customization/installation.md:107]: `{file}` renders with semantic HTML to designate it as a file, as opposed to an inline literal. I'd suggest doing that wherever you have files.

Also use care to capitalize Volto as needed.

```suggestion
You might have noticed that we have {file}`volto.config.js` in the root of the project. This is the Volto configuration file allowing us to configure Volto and register add-ons. The add-ons list points to the add-on we just installed. Cookieplone takes care of registering the add-on for us.
```

File: docs/volto_customization/installation.md
Comment (stevepiercy) [inline docs/volto_customization/installation.md:123]: Directories should also be marked with `{file}` in narrative text.

```suggestion
All the packages in the {file}`packages` directory will be included in the workspace.
```

File: docs/volto_customization/installation.md
Comment (stevepiercy) [inline docs/volto_customization/installation.md:136]:
```suggestion
We don't need to pin a specific workspace before we publish or release our project. pnpm takes care of dynamically updating the versions of these packages when you do `pnpm pack` or `pnpm publish`.
```

File: docs/volto_customization/installation.md
Comment (stevepiercy) [inline docs/volto_customization/installation.md:165]: Its a proper noun. Replace this and all other instances to "TypeScript".

```suggestion
### TypeScript configuration
```

File: docs/volto_customization/schema.md
Comment (stevepiercy) [inline docs/volto_customization/schema.md:187]:
```suggestion
Right now this variation only shows the image top variation of the Teaser block.
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
This looks really good. There a couple of naming things that need care, otherwise just trivial spelling and grammar. Thank you!

File: docs/volto_customization/installation.md
Comment (nileshgulia1) [reply docs/volto_customization/installation.md]: done [6e843e8](https://github.com/plone/training/pull/860/commits/6e843e8662610c197ab51afbdecf96bda948efc8)

File: docs/volto_customization/custom_block.md
Comment (stevepiercy) [inline docs/volto_customization/custom_block.md:12]: I totally misinterpreted this as being an inline literal to insert into a block. Sorry about the mistake.

Also I don't know why you use HTML markup when Markdown is easier, unless you want to force participants to open links in a new window with `target="_blank"`. This is the only training that does that. For this link and all others, if you want to use Markdown syntax, the pattern is `[label](URL)`.

```suggestion
Let's create a volto block that will display weather information for Brasilia. For this we can use [Open-Meteo API](https://open-meteo.com/). Open-Meteo is an open-source weather API and offers free access for non-commercial use. No API key required.
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
I goofed.

File: docs/volto_customization/custom_block.md
Comment (nileshgulia1) [reply docs/volto_customization/custom_block.md:12]: Sorry, My bad. There're also other dangling links since last time. I'll remove them. I agree markdown is better.

Comment (nileshgulia1) [reply docs/volto_customization/custom_block.md]: fixed in [a650478](https://github.com/plone/training/pull/860/commits/a650478b5d4c53a64428848e2936994cc45866c4)

**Comment by nileshgulia1** [comment]
@stevepiercy Could you please approve this if everything looks okay? Apologies for the delay—I've been busy finishing up tasks before the conference. Thank you!

**Comment by stevepiercy** [comment]
@nileshgulia1 sorry, I didn't realize you were done. I'll start a review tonight.

**Review by stevepiercy — APPROVED** [review]
I won't have time to do an in-depth review, but overall this LGTM.

I saw a few minor details. It's up to you if you want to search for and apply them.

- Volto should be capitalized when it is used in narrative text. In code and inline literals it depends on its usage.
- Add-on should be hyphenated in narrative text.
- Files and folders should have the `{file}` role applied, except when they are the label of a link, in which case they should be surrounded only by single backticks so that it visually appears similar to a file or folder.
- Inline literals should be surrounded by single backticks, such as `schemaEnhanced` or "`variations` key".

Thank you!

**Review by claudiaifrim — APPROVED** [review]

**Comment by nileshgulia1** [comment]
@stevepiercy Unfortunately I'm not authorized to merge this PR :(.

**Comment by stevepiercy** [comment]
@nileshgulia1 oh, my, that's not right. Invite sent! Please accept, and let me know if you can merge it now.

**Comment by nileshgulia1** [comment]
@stevepiercy Thanks! 🚀

**Pull request merged**
